base/path.jl

```diff
@@ -246,8 +246,70 @@ function splitpath(p::String)
     return out
 end
 
+joinpath(path::AbstractString) = path
```

Comment:
Might as well make the function type stable while you're at it and always return `String`.